src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SelectedMembers/AbstractSelectedMembers.cs

```diff
@@ -81,29 +89,20 @@ internal abstract class AbstractSelectedMembers<
             for (var i = fieldIndex; i < members.Count; i++)
             {
                 var member = members[i];
-                AddSelectedFieldOrPropertyDeclarations(member);
+                AddSelectedMemberDeclarations(member, membersToKeep);
             }
 
             return selectedMembers.ToImmutableAndFree();
```

Comment:
```suggestion
            return selectedMembers.ToImmutable();
```

this ensures we return teh pooled arary even if we end up cancelling the work we do.